crates/core/tedge_mapper/src/c8y/mapper.rs

```diff
@@ -64,115 +67,50 @@ impl TEdgeComponent for CumulocityMapper {
     async fn init(&self, cfg_dir: &Path) -> Result<(), anyhow::Error> {
         info!("Initialize tedge mapper c8y");
         create_directories(cfg_dir)?;
+
+        // HIPPO: Are these subscriptions still needed on init?
```

Comment:
It was the latter: `the idea of creating a session with --init`. Why not have this session created on the very startup of `tedge-mapper c8y`? What benefit do we get by creating this session while installing the plugin itself? If not, can we remove it?

Reply:
> Why not have this session created on the very startup of tedge-mapper c8y?

This is already the case. On start the mapper creates a connection with its session name, no clear flag and its subscriptions (exactly as done by `--init`). Calling `--init` is not required for the mapper to have its session created.

> What benefit do we get by creating this session while installing the plugin itself? 

This was to fix issues with the mapper and the agent sending messages to each others on start.

Currently, if you don't call `--init` just after the very *first* installation of thin-edge, then start the mapper and then the agent; you will get into the bug the software list is not sent to c8y. The message sent by the mapper to the agent being not received by the latter because published on a topic with no subscriptions *yet*. Similarly messages published on mapper's topics, before the mapper as even run once, are lost.

> If not, can we remove it?

Yes, we can.

Instead of blindly sending message on start, the mapper, the agent and all thin-edge components must await that the peer is up listening to the appropriate `tedge/health/#` topics.

> Obviously there's no harm in keeping it other than some minor consistency issues like the initial session may not have all the subscription topics derived from custom operations etc as most of these custom ops may not even be installed at the time of mapper installation.

There is a drawback of using `--init`. I notice that many are running `--init` when something is going unexpected. But `--init` is not idempotent nor without effects. If the session where already created, the pending messages are consumed and lost. 

> So, it might be better to have this session created on the very first mapper startup (tedge connect c8y) when typically all other ops are also in place.

This is already done.

> I'm just trying to remove as much as possible from --init blocks so that eventually we can get rid of it completely, if possible.

Yes, this would be great. I would like deprecate all these `--init` functions.

Reply:
Deferring this as it can't be done until the dependency between mapper and the agent isn't rewritten using health checks.